app/workers/concerns/dump_worker.rb

```diff
@@ -40,16 +46,14 @@ def dump_type
     "project_#{dump_target}_export"
   end
 
-  def temp_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}"
-  end
-
   def csv_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.csv"
+    @csv_tempfile ||= Tempfile.new(['export', '.csv'])
+    @csv_tempfile.path
   end
 
   def gzip_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.gz"
+    @gzip_tempfile ||= Tempfile.new(['export', '.gz'])
+    @gzip_tempfile.path
   end
 
   def project_file_path
```

Comment:
This should probably replace all non-alphanum characters rather than just whitespace (i.e. change \s to \W).